### DISCUSSION TITLE
Conditional content dont charge alpine

### DISCUSSION THREAD

**Discussion by SamuelBoudine** [livewire/livewire#5186] [1. Help]
Hello, so voila, I have a livewire component that contains an alpine.js component in an @if. My problem is that my x-data in my @if doesn't load even after my @if is true.

speed exemple : 

```html

@if($song)
       <div x-data="songData">
            @push('scripts')
        <script defer>
            document.addEventListener('alpine:init', () => {
                Alpine.data('songData', () => ({
                    variable1: 'hello',
                    fnnumero1(name) {
                        this.variable1 = name;
                    },
                }))
            })
        </script>
          ...
    @endpush
       </div>
@endif
```
error : Alpine Expression Error: songData is not defined

**Comment by joshhanley** [comment]
@SamuelBoudine yeah that’s correct. Livewire doesn’t support dynamically adding to the scripts stack on subsequent requests. So you will need to make sure to move your script out of the conditional so it’s rendered on the page initially. 

We’re working on support for it, but ran into some breaking changes. So hopefully we will get it into the next major version. 

Hope this helps!
